WalletWasabi/WabiSabi/Backend/Rounds/Round.cs

```diff
@@ -16,33 +18,30 @@ public Round(RoundParameters roundParameters)
 		{
 			RoundParameters = roundParameters;
 
-			AmountCredentialIssuer = new(new(Random), Random, MaxRegistrableAmount);
-			VsizeCredentialIssuer = new(new(Random), Random, PerAliceVsizeAllocation);
+			AmountCredentialIssuer = new(new(RoundParameters.Random), RoundParameters.Random, MaxRegistrableAmount);
+			VsizeCredentialIssuer = new(new(RoundParameters.Random), RoundParameters.Random, PerAliceVsizeAllocation);
 			AmountCredentialIssuerParameters = AmountCredentialIssuer.CredentialIssuerSecretKey.ComputeCredentialIssuerParameters();
 			VsizeCredentialIssuerParameters = VsizeCredentialIssuer.CredentialIssuerSecretKey.ComputeCredentialIssuerParameters();
 
 			var allowedAmounts = new MoneyRange(roundParameters.MinRegistrableAmount, RoundParameters.MaxRegistrableAmount);
 			var txParams = new MultipartyTransactionParameters(roundParameters.FeeRate, allowedAmounts, allowedAmounts, roundParameters.Network);
 			CoinjoinState = new ConstructionState(txParams);
 
-			Hash = new(HashHelpers.GenerateSha256Hash($"{Id}{MaxInputCountByAlice}{MinRegistrableAmount}{MaxRegistrableAmount}{PerAliceVsizeAllocation}{AmountCredentialIssuerParameters}{VsizeCredentialIssuerParameters}{FeeRate.SatoshiPerByte}"));
+			Id = CalculateHash();
```

Comment:
Why is this better than the original one?

Reply:
There are two things here:

* The hash is now calculated with Strobe what makes it resistant to manipulations like injection attacks.
* The Id is now the hash itself because the hash is the fingerprint of the instance and two instances can have the same id only if they share the same data. Given the hash is the id we can remove one.

Reply:
Nice